### PR TITLE
Remove real relation in non-discrete-time controlled if-statement.

### PR DESCRIPTION
**Pull request by qlambert-pro**
According to Section 3.8.5
real elementary relations inside if-clauses controlled by a non-discrete-time switching expression are not allowed.

### CODE REVIEW

File: Modelica/Blocks/Sources.mo [hunk 1]
```diff
@@ -1443,8 +1443,8 @@ a flange according to a given acceleration.
       b := b - a*shiftTimeScaled;
     end getInterpolationCoefficients;
   algorithm
-    if noEvent(size(table, 1) > 1) then
-      assert(not (table[1, 1] > 0.0 or table[1, 1] < 0.0), "The first point in time has to be set to 0, but is table[1,1] = " + String(table[1, 1]));
+    if size(table, 1) > 1 then
+      assert(noEvent(not (table[1, 1] > 0.0 or table[1, 1] < 0.0)), "The first point in time has to be set to 0, but is table[1,1] = " + String(table[1, 1]));
```
Comment (HansOlsson) [inline Modelica/Blocks/Sources.mo:1447]:
```suggestion
      assert(not (table[1, 1] > 0.0 or table[1, 1] < 0.0), "The first point in time has to be set to 0, but is table[1,1] = " + String(table[1, 1]));
```
No need for noEvent here either.

**Review by HansOlsson — APPROVED** [review]
Technically, these aren't forbidden as `table` is a parameter, and `noEvent` only matters if the operands are non-discrete-time expressions.

However, that makes it confusing and redundant so it would be good to remove them.

**Comment by qlambert-pro** [comment]
> if the operands are non-discrete-time expressions.

The specs require a touch up then.

**Comment by Harisankar-Allimangalath** [comment]
@HansOlsson  can you suggest or add anyone else also for reviewing whom you think would be a subject matter expert , Thanks .

**Comment by HansOlsson** [comment]
> @HansOlsson can you suggest or add anyone else also for reviewing whom you think would be a subject matter expert , Thanks .

As far as I understand the library officers for Modelica.Blocks are @AHaumer and @MartinOtter but this isn't needing any special Blocks-competence.

**Comment by Harisankar-Allimangalath** [comment]
Thanks @HansOlsson for clarifying . Let me add @hubertus65  for reviewing for the sake of merging (as it demands 2 reviewer's approval). Thankyou

**Review by hubertus65 — APPROVED** [review]

**Comment by henrikt-ma** [comment]
> Technically, these aren't forbidden as `table` is a parameter, and `noEvent` only matters if the operands are non-discrete-time expressions.
> 
> However, that makes it confusing and redundant so it would be good to remove them.

I agree that it is confusing.  Would it be desirable to get a warning when `noEvent` is applied to a parameter expression?  What about the more general case of `noEvent(…)` not being a non-discrete-time expression?

**Review by MartinOtter — APPROVED** [review]

**Comment by hubertus65** [comment]
Note from the MAP-LIB meeting: providing a warning about unnecessary and no-effect noEvent() operators is a quality of implementation issue. Warnings would be appreciated by modelers, but are not considered mandatory. 2023-11-14. Ticket is considered ready to merge.

**Pull request merged**